### PR TITLE
added sieve example for subaddress sorting

### PR DESCRIPTION
**Pull request by mpldr**
# Description

This just adds an example on how to automatically sort incoming mails, based on their subaddresses. I use this filter myself and thought it might be of interest to people setting up their sieves.

## Type of change

- [x] Documentation update

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by casperklein — APPROVED** [review]
Good contribution 👍

**Review by wernerfred — APPROVED** [review]
Thank you for your first contribution 🚀 

One question: Will this also work if you use another delimiter or would you need to adjust something?

**Comment by mpldr** [comment]
On Tue Feb 15, 2022 at 9:05 PM CET, Frederic Werner wrote:
> One question: Will this also work if you use another delimiter or would you need to adjust something?
If you use a different path separator, you need to change that.
--
Moritz Poldrack
https://moritz.sh

**Comment by mpldr** [comment]
I am also not aware of a way to automatically switch.

**Comment by wernerfred** [comment]
I do not mean the delimiter option on the dms in general but in your sieve code snippert. Can't see where it matches to `+` (which could be changed). Or will this sieve snippet also work if someone configured dms to use another delimiter out of the box

**Comment by mpldr** [comment]
On that I am actually not too sure. If I read the docs correctly, the
`require subaddress` *should* take care of it.

File: docs/content/config/advanced/mail-sieve.md [hunk 1]
```diff
@@ -61,6 +61,23 @@ pipe "external-program";
 
 For more examples or a detailed description of the Sieve language have a look at [the official site](http://sieve.info/examplescripts). Other resources are available on the internet where you can find several [examples](https://support.tigertech.net/sieve#sieve-example-rules-jmp).
 
+## Automatic sorting based on subaddresses
```
Comment (georglauterbach) [inline docs/content/config/advanced/mail-sieve.md:64]: Just a small nitpick: we usually use uppercase headings, so should would in fact be `## Automatic Sorting Based on Subaddresses` :)

**Review by georglauterbach — APPROVED** [review]

**Review by casperklein — APPROVED** [review]

**Pull request merged**